src/librustc_mir/interpret/place.rs

```diff
@@ -276,6 +276,10 @@ impl<Tag: ::std::fmt::Debug> Place<Tag> {
 }
 
 impl<'tcx, Tag: ::std::fmt::Debug> PlaceTy<'tcx, Tag> {
+    pub fn null(cx: &impl HasDataLayout, layout: TyAndLayout<'tcx>) -> Self {
```

Comment:
This is an awful hack that we do not want to perpetuate... at the very east, please make this `pub(super)` and add a doc comment that it should not be used and exists as temporary hack.

NULL places should just never exist, as a basic invariant on places.

Reply:
This was actually left over from my previous approach, so I removed all the `null` APIs

Reply:
Ah right you can get rid of them, that's awesome. :D